packages/server/src/routes/index/login.ts

```diff
@@ -28,9 +31,27 @@ router.get('login', async (_path: SubPath, ctx: AppContext) => {
 	if (ctx.joplin.owner) {
 		return redirect(ctx, homeUrl());
 	}
+
+	if (config().disableBuiltinLoginFlow) {
+		return await generateRedirectHtml('web-login');
+	}
+
 	return makeView();
 });
 
+// Log in using external authentication.
+router.get('login/:id', async (path: SubPath, ctx: AppContext) => {
+	if (ctx.joplin.owner) { // Already logged-in
+		return redirect(ctx, homeUrl());
+	} else if (config().saml.enabled && path.id === 'sso-saml') { // Server page, SAML
+		return await generateRedirectHtml('web-login');
+	} else if (config().saml.enabled && path.id === 'sso-saml-app') { // Client, SAML
+		return await generateRedirectHtml('app-login');
+	} else {
+		return makeView();
+	}
+});
```

Comment:
That's only relevant for SAML, is that correct? In that case, it should throw an error at the top if SAML is not enabled